uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -117,6 +117,14 @@ protected function update($params) {
       );
     }
 
+    if ($this->isTOILWithPastDates($params) && $this->isCancelledStatus($params)) {
+      if (!$this->canCanCancelTOILWithPastDates($params)) {
```

Comment:
Maybe instead of nested `if`s you could use variables to store the conditions? Just seems like there's no point in a nested `if` with no `else`